syncthing-git/PKGBUILD

```diff
@@ -5,63 +5,66 @@
 # Previous Maintainer : Sauyon Lee <sauyonl@sauyon.com>
 # Contributor : Martin Wimpress <code@flexion.org>
 
-pkgname=syncthing-git
-pkgver=v1.5.0.rc.2.r18.g037934ec7
+_pkgname=syncthing
+pkgname=$_pkgname-git
+pkgver=v1.6.0.rc.4.r21.g668979605
 pkgrel=1
 pkgdesc="Open Source Continuous Replication / Cluster Synchronization Thing"
-url="http://syncthing.net/"
+url="https://syncthing.net/"
 license=('MPLv2')
 makedepends=('git' 'go' 'inetutils')
-conflicts=('syncthing')
-provides=('syncthing')
+conflicts=("$_pkgname")
+provides=("$_pkgname")
 arch=('i686' 'x86_64')
-source=("$pkgname::git+https://github.com/syncthing/syncthing.git")
+source=("git+https://github.com/syncthing/syncthing.git")
 sha256sums=('SKIP')
-install=${pkgname}.install
-_name=syncthing
+install=${_pkgname}.install
 
 prepare() {
-	cd "${srcdir}"
+	cd "$srcdir/$_pkgname"
+	mkdir -p "${srcdir}/../go"
+	export GOPATH="${srcdir}/../go"
+	export PATH="$PATH:$GOPATH/bin"
+	# fix for broken version
+	chmod u+w -R "$GOPATH"
+	go get -d -v ./...
 }
 
 pkgver() {
-	cd "${pkgname}"
+	cd "$srcdir/$_pkgname"
 	git describe --long --tags | sed 's/\([^-]*-g\)/r\1/;s/-/./g'
 }
 
 build() {
-	export GOPATH="${srcdir}/${pkgname}-${pkgver}"
-	cd "${srcdir}"
-	mkdir -p "src/github.com/syncthing"
-	mv "${pkgname}" "src/github.com/syncthing/${_name}"
-	cd "src/github.com/syncthing/${_name}"
-	if [ ${CARCH}" == "i686" ] ; then
-			go run build.go -no-upgrade -goarch 386 build
-	if [ ${CARCH}" == "x86_64" ] ; then
-			go run build.go -no-upgrade -goarch amd64 build
-	else
-			go run build.go -no-upgrade build
-	fi
+	cd "$srcdir/$_pkgname"
+	export GOPATH="${srcdir}/../go"
+	export PATH="$PATH:$GOPATH/bin"
+	export CGO_LDFLAGS="${LDFLAGS}"
+	export CGO_CFLAGS="${CFLAGS}"
+	export CGO_CPPFLAGS="${CPPFLAGS}"
+	export CGO_CXXFLAGS="${CXXFLAGS}"
+	export GOFLAGS="-buildmode=pie -trimpath -mod=readonly -modcacherw"
+	go run build.go -no-upgrade build
 }
 
 check() {
-	export GOPATH="${srcdir}"
-	cd "${srcdir}/src/github.com/syncthing/${_name}"
-#	 go run build.go -no-upgrade test
+	export GOPATH="${srcdir}/../go"
+	cd "$srcdir/$_pkgname"
+#	go run build.go -no-upgrade test
 }
 
 package() {
-	cd "${srcdir}/src/github.com/syncthing/${_name}"
-	install -Dm755 ${_name} "${pkgdir}/usr/bin/${_name}"
-	install -Dm644 README.md "${pkgdir}/usr/share/doc/${_name}/README.md"
-	install -Dm644 "etc/linux-systemd/system/${_name}@.service" "${pkgdir}/usr/lib/systemd/system/${_name}@.service"
-	install -Dm644 "etc/linux-systemd/user/${_name}.service" "${pkgdir}/usr/lib/systemd/user/${_name}.service"
+	cd "$srcdir/$_pkgname"
+	install -Dm755 ${_pkgname} "${pkgdir}/usr/bin/${_pkgname}"
+	install -Dm644 README.md "${pkgdir}/usr/share/doc/${_pkgname}/README.md"
+	install -Dm644 "etc/linux-systemd/system/${_pkgname}@.service" "${pkgdir}/usr/lib/systemd/system/${_pkgname}@.service"
```

Comment:
tbh I would create a system user and modify the service file/create your own to run just as this user. This way you can use way more hardening options.

Take a look:

https://github.com/ipfs/go-ipfs/pull/7286

Reply:
Thanks a lot, I may do this at a later time but not now.